app/src/main/kotlin/com/wire/android/notification/broadcastreceivers/CallNotificationDismissReceiver.kt

```diff
@@ -0,0 +1,64 @@
+package com.wire.android.notification.broadcastreceivers
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.wire.android.di.KaliumCoreLogic
+import com.wire.android.notification.CallNotificationManager
+import com.wire.android.util.dispatchers.DispatcherProvider
+import com.wire.kalium.logic.CoreLogic
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.id.toConversationId
+import com.wire.kalium.logic.feature.session.CurrentSessionResult
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class CallNotificationDismissReceiver : BroadcastReceiver() {
+
+    @Inject
+    @KaliumCoreLogic
+    lateinit var coreLogic: CoreLogic
+
+    @Inject
+    lateinit var dispatcherProvider: DispatcherProvider
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val conversationId: String = intent.getStringExtra(EXTRA_CONVERSATION_ID) ?: return
+
+        val userId: QualifiedID? =
+            intent.getStringExtra(EXTRA_RECEIVER_USER_ID)?.toConversationId() //TODO bad naming, need to be toQualifiedID()
+
+        GlobalScope.launch(dispatcherProvider.io()) {
+            val sessionScope =
+                if (userId != null) {
+                    coreLogic.getSessionScope(userId)
+                } else {
+                    val currentSession = coreLogic.getAuthenticationScope().session.currentSession()
+                    if (currentSession is CurrentSessionResult.Success) {
+                        coreLogic.getSessionScope(currentSession.authSession.userId)
+                    } else {
+                        null
+                    }
+                }
+
+            sessionScope?.let {
+                it.calls.rejectCall(conversationId.toConversationId())
+                CallNotificationManager.cancelNotification(context)
```

Comment:
Shouldn't we cancel the notification even if there is no sessionScope? It shouldn't happen but if it does, then I think we don' want such notification to keep hanging then 😄

Reply:
good point! :)